pandas/plotting/_core.py

```diff
@@ -2835,18 +2835,42 @@ def area(self, x=None, y=None, **kwds):
 
     def pie(self, y=None, **kwds):
         """
-        Pie chart
+        Generate a pie plot.
+
+        A pie plot is a representation of the distribution of numerical data
+        in a DataFrame column. This function wraps the `matplotlib.pyplot.pie`
+        function for specified column. If no column reference is passed and
+        ``subplots`` argument is set to ``True``, an np.array of plots for
+        all DataFrame columns will be returned.
 
         Parameters
         ----------
-        y : label or position, optional
-            Column to plot.
-        `**kwds` : optional
+        y : str or int, optional
+            Label or position of the column to plot.
+            If not provided, ``subplots=True`` argument must be passed.
+        **kwds : optional
             Keyword arguments to pass on to :py:meth:`pandas.DataFrame.plot`.
 
         Returns
         -------
-        axes : matplotlib.AxesSubplot or np.array of them
+        axes : matplotlib.AxesSubplot or np.array of them.
+
+        See Also
+        --------
+        :meth:`pandas.Series.plot.pie` : Generate a pie plot for a Series.
+
+        Examples
+        --------
+        In the example below we have a DataFrame with the information about
```

Comment:
Perhaps a matter of style, but I'd say:
```
Given a DataFrame with information about some planets mass and radius, create a pie plot of their masses:
```
Because the fact that it is an example can be determined from the section it is in, and the fact that we pass the 'mass' column to the `pie` method can be seen in the code itself.

What about another example without specifying the `y` argument?